### PR TITLE
feat: add CODEOWNERS and fix blog page error

### PR DESCRIPTION
**Pull request by s2sharpit**
## Related Issue

- Information about the related issue

Closes: #403 

### Describe the changes you've made

Hey @vinzvinci @Anmol-Baranwal @Panquesito7 
I have added a sample file for the codeowners.
Let's work together to add the specific members as codeowners to specific task.

## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update.
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Screenshots (if applicable)

|      Original       |      Updated       |
| :-----------------: | :----------------: |
| original screenshot | updated screenshot |

## Code of Conduct

- [ ] I agree to follow this project's [Code of Conduct](https://github.com/WebXDAO/.github/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .github/CODEOWNERS
Comment (Panquesito7) [inline .github/CODEOWNERS:14]: I'd like to be added to be codeowner of MD files.

**Review by Anmol-Baranwal — CHANGES_REQUESTED** [review]
Please add the following files as well:

- `.gitignore`
- `.github/workflows/`
- `src/app`
- `src/components`
- `.github`

@Panquesito7 
What are your thoughts? Do you think different people should be responsible for reviewing different parts of the `src` directory?

**Comment by s2sharpit** [comment]
> Please add the following files as well:
> 
> * `.gitignore`
> * `.github/workflows/`
> * `src/app`
> * `src/components`
> * `.github`
> 
> @Panquesito7 What are your thoughts? Do you think different people should be responsible for reviewing different parts of the `src` directory?

Dear @vinzvinci @Panquesito7 @mkubdev @Anmol-Baranwal 

Thank you for reaching out and providing the list of files to be added. To clarify the code ownership for the files mentioned, could you please specify which individuals or teams should be responsible for reviewing each file? It would be helpful to have a clear understanding of the desired ownership assignments within the src directory.

Once you provide the necessary details, we can proceed with assigning the appropriate code owners to ensure efficient and effective code review.

Looking forward to your response.

**Comment by vinzvinci** [comment]
`@WebXDAO/webx-guild-developer` is enough in the long run

**Comment by vinzvinci** [comment]
Hi @Panquesito7 @Anmol-Baranwal we need to move forward on this

**Comment by Anmol-Baranwal** [comment]
> Hi @Panquesito7 @Anmol-Baranwal we need to move forward on this

@s2sharpit 
Please make the change, and `@WebXDAO/webx-guild-developer` will be responsible for every change along with the existing core team.

Make the change, and we will review this very fast.

**Comment by s2sharpit** [comment]
> > Hi @Panquesito7 @Anmol-Baranwal we need to move forward on this
> 
> @s2sharpit Please make the change, and `@WebXDAO/webx-guild-developer` will be responsible for every change along with the existing core team.
> 
> Make the change, and we will review this very fast.

Hey @Anmol-Baranwal 
I have made the changes and added `@WebXDAO/webx-guild-developer` & `@WebXDAO/webx-core` as responsible for every change in the project.

**Review by Anmol-Baranwal — CHANGES_REQUESTED** [review]
@s2sharpit 
Having both `* @WebXDAO/webx-guild-developer` and `/src/* @WebXDAO/webx-guild-developer` in the CODEOWNERS file would be redundant. The `*` wildcard already covers all files and directories in the repository, so the `/src/*` pattern is unnecessary.

**Comment by Anmol-Baranwal** [comment]
@vinzvinci 
Please check, why the deployment failed and please review the changes.

**Comment by s2sharpit** [comment]
> @s2sharpit Having both `* @WebXDAO/webx-guild-developer` and `/src/* @WebXDAO/webx-guild-developer` in the CODEOWNERS file would be redundant. The `*` wildcard already covers all files and directories in the repository, so the `/src/*` pattern is unnecessary.

Thank you for pointing that out! I've now removed the redundant /src/* pattern from the CODEOWNERS file.

**Comment by s2sharpit** [comment]
> @vinzvinci Please check, why the deployment failed and please review the changes.

Hey @vinzvinci and @Anmol-Baranwal,

I discovered an error in the data fetched from the API on the blog page. It seems that the data entry with the ID `1537068` is missing the `cover_image` URL, resulting in it being `null` by default. It appears that this issue may have arisen due to recent additions to the data points in the API.

Screenshot

To address this situation, I've made a simple modification in the code: `src={curElem.cover_image || ""}`

Please take a look at the code changes and review the recent modifications.

**Comment by Anmol-Baranwal** [comment]
It is recommended to work on a single issue at a time, such as `codeowners`, because working on the blog page should be done in a new PR. So, please don't make any new changes since it takes time to review those changes.

**Review by mkubdev — CHANGES_REQUESTED** [review]
Hi, codeowner file syntax look good to me!
Kindly ask you to add the requested change about checking if the image source isavailable before rendering the component, and provide an `alt={curElem.title}` for the images. 😄

File: src/app/blogs/Blog.tsx
Comment (darkterminal) [reply src/app/blogs/Blog.tsx:42]: Use the default image from `https://og-image-rest-generator.fly.dev/seo-banner` with this example structure `https://og-image-rest-generator.fly.dev/seo-banner?title=WebXDAO%20x%20Software%20Freestyle%20Engineer&author=darkterminal&head=Storyteller&writer=Punk%20Storyteller&logo=https://avatars.githubusercontent.com/u/89759498?s=200&v=4` if cover image is `null`

The result is:
WebXDAO

**Comment by s2sharpit** [comment]
There are some errors. Will fix it after some time. I'm a lit bit busy.

File: src/app/blogs/Blog.tsx
Comment (s2sharpit) [reply src/app/blogs/Blog.tsx]: Hey @mkubdev,

I wanted to let you know that I've already added the alt attribute to the images as you suggested. Thanks for bringing it up! However, upon further consideration, I realized that removing the image when it's not available may cause a layout shift, which could negatively impact the UI's appearance.

In light of this, I fully endorse @darkterminal's proposal. He suggested using a default image from https://og-image-rest-generator.fly.dev/seo-banner if the cover image is missing. I find this idea brilliant as it ensures that the UI remains visually appealing even in the absence of a cover image.

Thus, I'm proceeding to implement both your suggestions, along with the addition of the default image option. Your input is highly appreciated, and together, let's enhance this feature even more! 😄

**Comment by Anmol-Baranwal** [comment]
> There are some errors. Will fix it after some time. I'm a lit bit busy.

It's just a minor issue.

> Repeated code with wrong block structure
image

**Comment by s2sharpit** [comment]
Hi @Anmol-Baranwal, there's one more issue we need to address. When we remove the image, the layout shifts and the UI doesn't look good. I propose we follow the suggestion made by @darkterminal in [this message](https://github.com/WebXDAO/WebXDAO.github.io/pull/474#discussion_r1271426877) to fix this problem.

Could you please share your thoughts on this suggestion? Thank you!

**Comment by Anmol-Baranwal** [comment]
> Hi @Anmol-Baranwal, there's one more issue we need to address. When we remove the image, the layout shifts and the UI doesn't look good. I propose we follow the suggestion made by @darkterminal in [this message](https://github.com/WebXDAO/WebXDAO.github.io/pull/474#discussion_r1271426877) to fix this problem.
> 
> Could you please share your thoughts on this suggestion? Thank you!

Yeah, incorporate that if it fixes that issue.

**Comment by s2sharpit** [comment]
> > Hi @Anmol-Baranwal, there's one more issue we need to address. When we remove the image, the layout shifts and the UI doesn't look good. I propose we follow the suggestion made by @darkterminal in [this message](https://github.com/WebXDAO/WebXDAO.github.io/pull/474#discussion_r1271426877) to fix this problem.
> > Could you please share your thoughts on this suggestion? Thank you!
> 
> Yeah, incorporate that if it fixes that issue.

Hello @Anmol-Baranwal, I've implemented it to the best of my knowledge.  If any changes are needed, just let me know.

**Comment by Anmol-Baranwal** [comment]
I will review this as soon as possible.

File: src/app/blogs/Blog.tsx
Comment (darkterminal) [inline src/app/blogs/Blog.tsx:49]: This is an interesting point for me, but it would be nice if this was a helper and not written in a crude way. :thinking:

**Review by darkterminal — CHANGES_REQUESTED** [review]

File: src/app/blogs/Blog.tsx
Comment (s2sharpit) [reply src/app/blogs/Blog.tsx:49]: Hi @darkterminal, thanks for your feedback! I appreciate your input. Could you kindly clarify what you mean by "helper" and "crude way"? I'd love to understand your perspective better.

Comment (darkterminal) [reply src/app/blogs/Blog.tsx:49]: Create separate reusable functions called `createDefaultImageCover` (like) in `src/lib/utils.ts` that are responsible for his behavior. You can be defined the following parameters that represent the query params in that URL.

Comment (s2sharpit) [reply src/app/blogs/Blog.tsx:49]: Okay, Sure

Comment (s2sharpit) [reply src/app/blogs/Blog.tsx]: Hey @darkterminal, I have made the suggested changes. Please review them.

Comment (Anmol-Baranwal) [reply src/app/blogs/Blog.tsx]: I am resolving the conversation, since darkterminal approved the changes.

**Comment by s2sharpit** [comment]
Hey @Anmol-Baranwal,
Could you please review this PR at your earliest convenience?

**Comment by Anmol-Baranwal** [comment]
> Hey @Anmol-Baranwal, Could you please review and merge this PR at your earliest convenience?

Without proper review and confirmation from others, I cannot merge it.

If you are worried about GSSoC'23, then don't be, because we will be merging it before that.

**Review by Anmol-Baranwal — APPROVED** [review]
Nicely done! Everything looks good to me 👍

**Review by mkubdev — APPROVED** [review]
Good job! 👏 Thanks a lot for your contribution. Everything look good to me 👍🏻

**Comment by s2sharpit** [comment]
Hey @mkubdev @Anmol-Baranwal,
Can you merge this now? Worried about potential conflicts later.

**Pull request merged**
